ang/test/civicaseextras/CaseOutcome.spec.js

```diff
@@ -0,0 +1,199 @@
+            })
+            .map(function (value, fieldName) {
+              return {
+                label: customFieldsMockFieldsMap[fieldName].label,
+                value: value
+              };
+            })
+            .value();
+
+          return {
+            activityType: activityType,
+            customFields: customFields
+          };
+        });
+
+        initController();
+      });
+
+      it('requests a list of activity outcomes', function () {
+        expect(crmApi).toHaveBeenCalledWith('Activity', 'get', {
+          'sequential': 1,
+          'case_id': $scope.case.id,
+          'activity_type_id.grouping': 'outcome'
+        });
+      });
+
+      it('stores a list of activity outcomes and their custom fields', function () {
+        expect($scope.activityOutcomes).toEqual(expectedOutcomes);
+      });
+    });
+
+    describe('when the case details have been updated', function () {
+      beforeEach(function () {
+        initController();
+        $rootScope.$emit('updateCaseData');
+      });
+
+      it('refreshes the case outcome data', function () {
+        expect(crmApi).toHaveBeenCalledWith('Activity', 'get', {
+          'sequential': 1,
+          'case_id': $scope.case.id,
+          'activity_type_id.grouping': 'outcome'
+        });
+      });
+    });
+
+    /**
+     * Initializes the case outcome controller.
+     */
+    function initController () {
+      $scope = $rootScope.$new();
+      $scope.case = { id: _.uniqueId() };
+
+      $controller('civicaseExtrasCaseOutcomeController', { $scope: $scope });
+      $rootScope.$digest();
+    }
+  });
+
+  describe('CiviCaseExtrasCaseOutcome directive', function () {
+    var $compile, $q, $rootScope, $scope, activitiesMockData, caseOutcome, crmApi, masonryGrid;
+
+    beforeEach(module('civicaseextras', 'civicaseextras.templates', 'civicase.data', function ($provide, $compileProvider) {
+      masonryGrid = jasmine.createSpyObj('masonryGrid', ['addGridItemAt', 'removeGridItem']);
+      crmApi = jasmine.createSpy('crmApi');
+
+      $provide.value('crmApi', crmApi);
+      $compileProvider.directive('civicaseMasonryGrid', function () {
+        return {
+          controller: function () {
+            var vm = this;
+
+            vm.addGridItemAt = masonryGrid.addGridItemAt;
+            vm.removeGridItem = masonryGrid.removeGridItem;
+          }
+        };
+      });
+    }));
+
+    beforeEach(inject(function (_$compile_, _$q_, _$rootScope_, _activitiesMockData_) {
+      $compile = _$compile_;
+      $q = _$q_;
+      $rootScope = _$rootScope_;
+      activitiesMockData = _activitiesMockData_.get();
+
+      crmApi.and.returnValue($q.resolve({
+        count: activitiesMockData.length,
+        values: activitiesMockData
+      }));
+    }));
+
+    afterEach(function () {
+      $('[civicase-masonry-grid]').remove();
+    });
+
+    describe('interacting with the masonry grid container', function () {
```

Comment:
is this whole test suite necessary? Aren't you simply testing if the masonry grid directive behaves correctly? If that's the case, that's what the directive's original test suite is for, no? What's so special about the outcome panel that needs to be explicitly tested?

Reply:
No, this tests something different. The case outcome must be included as the first element of the grid and this is what the spec is testing.